packages/shared-ui/src/components/header/Header.tsx

```diff
@@ -0,0 +1,335 @@
+  }
+
+  return null
+}
+
+function HeaderImageElement({ ...props }: ImgHTMLAttributes<HTMLImageElement>) {
+  return (
+    <img
+      id="header-image"
+      src={props.src}
+      alt={props.alt || 'Arbitrum logo'}
+      className={twMerge('-ml-2 w-56 lg:ml-0 lg:w-60', props.className || '')}
+    />
+  )
+}
+
+export function HeaderContent({ children }: { children: React.ReactNode }) {
+  const mutationObserverRef = useRef<MutationObserver>()
+  const [, setMutationCycleCount] = useState(0)
+
+  useEffect(() => {
+    const header = document.getElementById('header')
+
+    if (!header) {
+      return
+    }
+
+    /**
+     * On mobile, the header opens up and closes through a popover component by clicking the hamburger menu.
+     * It's possible for the popover to be closed at the time of rendering this component.
+     * This means that the portal root element won't be found, and nothing will be rendered to the portal.
+     *
+     * This is a little trick that sets up a `MutationObserver` to listen for changes to the header subtree.
+     * Each time a mutation happens, it will call `setMutationCycleCount`, forcing a re-render to the portal.
+     *
+     * There's no real performance concern, as the contents of the header change very rarely.
+     */
+    const config = { subtree: true, childList: true }
+    mutationObserverRef.current = new MutationObserver(() =>
+      setMutationCycleCount(
+        prevMutationCycleCount => prevMutationCycleCount + 1
+      )
+    )
+    mutationObserverRef.current.observe(header, config)
+
+    return () => mutationObserverRef.current?.disconnect()
+  }, [])
+
+  const rootElement = document.getElementById('header-content-root')
+
+  if (!rootElement) {
+    return null
+  }
+
+  return ReactDOM.createPortal(children, rootElement)
+}
+
+export function Header(props: HeaderProps) {
+  return (
+    <header id="header" className={defaultHeaderClassName}>
+      <div className="flex w-full max-w-[1440px] justify-between px-8">
+        <div className="flex items-center lg:space-x-2 xl:space-x-12">
+          <a href="/" className="arb-hover flex flex-col items-center">
+            <HeaderImageElement src={props.logoSrc} />
+          </a>
+          <div className="hidden items-center lg:flex lg:space-x-2 xl:space-x-8">
+            <HeaderMenuDesktop {...learnMenuProps}>Learn</HeaderMenuDesktop>
+            <HeaderMenuDesktop
+              items={[
+                {
+                  title: 'App Portal',
+                  anchorProps: { href: 'https://portal.arbitrum.one' }
+                },
+                {
+                  title: 'Explorers',
+                  items: explorersMenuProps.items
+                }
+              ]}
+            >
+              Ecosystem
+            </HeaderMenuDesktop>
+            <HeaderMenuDesktop {...chartsStatsMenuProps}>
+              Charts & Stats
+            </HeaderMenuDesktop>
+            <DesktopExternalLink href={GET_HELP_LINK}>
```

Comment:
We can just add a new prop for `getHelpHref` to `Header` for now, I think this might change throughout UIs.